scripts/screeps_rl_experiment_card.py

```
@@ -821,27 +845,153 @@ def select_accepted_dataset_gate(gate_root: Path, gate_id: str | None = None) ->
+    if relative_path.name != DATASET_GATE_REPORT_FILENAME:
+        return False
+    if len(relative_path.parts) == 2:
+        return relative_path.parts[0] != "gate-data"
+    return (
+        len(relative_path.parts) == 3
+        and relative_path.parts[0] in {"rl-dataset-gates", "rl-control-loop"}
+        and relative_path.parts[1] != "gate-data"
+    )
+
+
+def is_acceptable_dataset_gate_report(payload: Any) -> bool:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `payload`

(ANN401)